os/arch/arm/src/amebad/amebad_spi.c

```diff
@@ -0,0 +1,1237 @@
+ *
+ ************************************************************************************/
+
+#ifndef CONFIG_SPI_EXCHANGE
+static void amebad_spi_recvblock(FAR struct spi_dev_s *dev, FAR void *rxbuffer,
+				size_t nwords)
+{
+	spiinfo("rxbuffer=%p nwords=%d\n", rxbuffer, nwords);
+	return amebad_spi_exchange(dev, NULL, rxbuffer, nwords);
+}
+#endif
+
+/************************************************************************************
+ * Name: amebad_spi_clock_enable
+ *
+ * Description:
+ *   Ungate SPI clock
+ *
+ ************************************************************************************/
+
+void amebad_spi_clock_enable(uint32_t base)
+{
+	return;
+}
+
+/************************************************************************************
+ * Name: amebad_spi_clock_disable
+ *
+ * Description:
+ *   Gate SPI clock
+ *
+ ************************************************************************************/
+
+void amebad_spi_clock_disable(uint32_t base)
+{
+	return;
+}
+
+/************************************************************************************
+ * Name: amebad_spi_bus_initialize
+ *
+ * Description:
+ *   Initialize the selected SPI bus in its default state (Master, 8-bit, mode 0, etc.)
+ *
+ * Input Parameters:
+ *   priv   - private SPI device structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+
+static void amebad_spi_bus_initialize(struct amebad_spidev_s *priv)
+{
+
+	DEBUGASSERT(priv);
+	DEBUGASSERT(&priv->spi_object);
+
+	priv->spi_object.spi_idx = priv->spi_idx;
+	spi_init(&priv->spi_object, priv->spi_mosi, priv->spi_miso, priv->spi_sclk, priv->spi_cs);
+
+	spi_format(&priv->spi_object, priv->nbits, priv->mode, priv->role);
+
+}
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: amebad_spibus_initialize
+ *
+ * Description:
+ *   Initialize the selected SPI bus
+ *
+ * Input Parameters:
+ *   Port number (for hardware that has mutiple SPI interfaces)
+ *
+ * Returned Value:
+ *   Valid SPI device structure reference on success; a NULL on failure
+ *
+ ************************************************************************************/
+
+FAR struct spi_dev_s *amebad_spibus_initialize(int bus)
+{
+	FAR struct amebad_spidev_s *priv = NULL;
+
+	irqstate_t flags = irqsave();
+
+	if (bus == 1) {
+		/* Select SPI1 */
+
+		priv = &g_spi0dev;
+
+		/* Only configure if the bus is not already configured */
+
+		amebad_spi_bus_initialize(priv);
+
+	} else if (bus == 2) {
+		/* Select SPI2 */
+
+		priv = &g_spi1dev;
+
+		/* Only configure if the bus is not already configured */
+
+		amebad_spi_bus_initialize(priv);
+
+	} else {
+		spierr("ERROR: Unsupported SPI bus: %d\n", bus);
+		return NULL;
```

Comment:
I think irqrestore(flags); is needed before return.

Reply:
Added irqrestore(flags); before return.